### ISSUE TITLE
Incorrect code gen

### ISSUE THREAD

**Issue by santolucito** [Barnard-PL-Labs/tsltools#70]
When testing this spec https://github.com/Barnard-PL-Labs/tsltools/blob/d2f61d1fd2a75d37fd3f510fa686d9a2e4dc4bf2/test/res/specs/Heating.tsl on https://barnard-pl-labs.github.io/tsl-api/ it generated the following code:

```
if (currentState === 0) {
  if (!room.heating.off() && room.heating.on() && (outside.temperature <= room.temperature)) {
    room.heating.ctrl = room.heating.ctrl
    currentState = 1
  }
```

The issue is that ```room.heating.off``` should be an input, but the generated code treated it as a function term with ```()``` at the end. probably a simple parsing issue, but need to fix the generation here.

Longer term, we should improve the tests...

**Comment by leoqiao18** [comment]
Fixed in #71 .

But yea, we should figure out how to properly test code gen output.

**Comment by leoqiao18** [comment]
Moving the discussion of codegen tests to #72
